app/pages/category/category.component.js

```diff
@@ -0,0 +1,37 @@
+class CategoryController {
+    constructor(appService, $state) {
+        'ngInject';
+        this.appService = appService;
+        this.$state = $state;
+    }
+
+    $onInit() {
+        this.getProducts();
+        this.data = [];
+        this.categoryData = [];
+        this.actualState = this.$state.params.id;
+    }
+
+    getProducts() {
+        this.appService.getProducts().then((data) => {
+            this.data = data;
+            this.getCategoryProducts(this.data, this.categoryData, this.actualState);
+        })
+    }
+
+    getCategoryProducts(data, categoryData, actualState) {
+        angular.forEach(data, function (param) {
```

Comment:
I would put as many logic in the services as I can - controller shouldn't have it too much